snap/snapcraft.yaml

```diff
@@ -108,19 +108,24 @@ apps:
     adapter: full
     after:
       - postgres
+      # to ensure kong.yml is ready for kong to process as the file is set up from security-secretstore-setup
+      - security-secretstore-setup
     command: bin/kong-daemon.sh
     command-chain:
       - bin/perl5lib-launch.sh
       - bin/kong-launch.sh
     daemon: forking
     environment:
       KONG_CONF: $SNAP_DATA/config/security-proxy-setup/kong.conf
+      KONGADMIN_CONFIGFILEPATH: $SNAP_DATA/config/kong.yml
       KONG_LOGS_DIR: $SNAP_COMMON/logs
       KONG_PROXY_ACCESS_LOG: $SNAP_COMMON/logs/kong-proxy-access.log
       KONG_ADMIN_ACCESS_LOG: $SNAP_COMMON/logs/kong-admin-access.log
       KONG_PROXY_ERROR_LOG: $SNAP_COMMON/logs/kong-proxy-error.log
       KONG_ADMIN_ERROR_LOG: $SNAP_COMMON/logs/kong-admin-error.log
       KONG_ADMIN_LISTEN: "localhost:8001, localhost:8444 ssl"
+      KONG_STATUS_LISTEN: "0.0.0.0:8100"
+      KONG_DNS_ORDER: "LAST,A,CNAME"
```

Comment:
not sure about the order, i just copied whatever has done in the compose file to the snap here.  @beaufrusetta maybe you can help to answer this better?

Reply:
Port 8100 doesn't need to be exposed externally (that is my fault) - just needs to be available to the local container/snap network in order for proxy-setup to verify that it is up and running so that it can continue setting up services/routes (proxy-setup was previously pinging the admin port to see if it was up, when it should have been using the status port all along). I have not tested binding it to the local Docker network, however, it **_should_** work. We should probably open up a different issue and I can quickly modify/fix that.

The DNS order was modified because of an issue with a Windows based host DNS issue. Keeping the default order would cause the first API request to the admin API to take min of 20 secs to complete, and then others would complete as expected. Over a certain amount of time, that would then reset, and the next API request coming in would exhibit the same behavior. Changing the order did not impact non-Windows based systems, so implementing it across the board seemed pretty low risk.

There are documented cases of this happening in other projects that utilize Kong containers on Windows hosts, and I verified that using my various machines. The work-around that was found in one of the documented issues was to remove the SRV record check by manipulating the Kong DNS checking order via environment variable. Once we did that, the issue went away.

(Thanks to @rsdmike for helping to expose and find a solution for that DNS issue)

Reply:
Actually - in the latest version of the `edgex-compose` code, I bound 8100 to the local network through the `ports` directive in the compose file. Setting `KONG_STATUS_LISTEN` is still set to 0.0.0.0:8100, however, Docker compose cuts off the ports access when it creates the container.

Reply:
thanks, @beaufrusetta . I"ll  try to test 8100 port on snap locally here to see if works.

Reply:
I just updated KONG_STATUS_LISTEN to use localhost in the snap and everything comes up just fine.

That said, I'm still seeing the 401 errors after a reboot. The failing service always seems to be different, sometimes its core-data, sometimes -metadata, -scheduler, ... The only consistent is it always seems to be the first that fails. 

I tried adding a 120s sleep before security-proxy-setup starts to rule out a race condition between security-secretstore-setup, and I still get the same failures.

Reply:
> I just updated `KONG_STATUS_LISTEN` to use localhost in the snap and everything comes up just fine.
> 
> That said, I'm still seeing the 401 errors after a reboot. The failing service always seems to be different, sometimes its core-data, sometimes -metadata, -scheduler, ... The only consistent is it always seems to be the first that fails.
> 
> I tried adding a 120s sleep before security-proxy-setup starts to rule out a race condition between security-secretstore-setup, and I still get the same failures.

yea, I also observed the same thing- it is really weird issue in snap. However, when I tried `snap stop` services first right after `snap install`, the 401 error seems gone away. So this is kind of unique in snap as I didn't see anything like this in docker-compose environment.   @tonyespy  any thoughts?

Reply:
security wg meeting discussion: we believe this is caused by running [initKongService](https://github.com/edgexfoundry/edgex-taf/issues/403#issuecomment-839338389) on an already initialized Kong

As a way to validate this is the error we could try querying the Kong admin api or run a kong config db_export maybe.

Are we erroniously starting proxy setup?

Reply:
added comments about `KONG_DNS_ORDER` based on @beaufrusetta 's feedback- thanks to @beaufrusetta !